acceptance/tests/terminating-gateway/terminating_gateway_destinations_test.go

```diff
@@ -0,0 +1,178 @@
+		},
+	}
+	for _, c := range cases {
+		name := fmt.Sprintf("secure: %t protcol: %s", c.secure, c.protocol)
+		t.Run(name, func(t *testing.T) {
+			ctx := suite.Environment().DefaultContext(t)
+
+			helmValues := map[string]string{
+				"connectInject.enabled":                    "true",
+				"terminatingGateways.enabled":              "true",
+				"terminatingGateways.gateways[0].name":     "terminating-gateway",
+				"terminatingGateways.gateways[0].replicas": "1",
+				"global.image":                             "dhayachi/consul:5aff2842051", // TODO: remove when 1.13 is released
+				"global.acls.manageSystemACLs":             strconv.FormatBool(c.secure),
+				"global.tls.enabled":                       strconv.FormatBool(c.secure),
+				"global.tls.autoEncrypt":                   strconv.FormatBool(c.secure),
+			}
+
+			logger.Log(t, "creating consul cluster")
+			releaseName := helpers.RandomName()
+			consulCluster := consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+			consulCluster.Create(t)
+			consulClient, _ := consulCluster.SetupConsulClient(t, c.secure)
+
+			// Deploy a static-server that will play the role of an external service.
+			logger.Log(t, "creating static-server deployment")
+			k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/bases/static-server-https")
+
+			// If ACLs are enabled we need to update the role of the terminating gateway
+			// with service:write permissions to the static-server service
+			// so that it can request Connect certificates for it.
+			if c.secure {
+				updateTerminatingGatewayRole(t, consulClient, terminatingGatewayRules)
+			}
+
+			// Since we are using the transparent kube DNS, disable the ability
+			// of the service to dial the server directly through the sidecar
+			createMeshConfigEntry(t, consulClient, "")
+
+			// Create the config entry for the terminating gateway.
+			createTerminatingGatewayConfigEntry(t, consulClient, "", "", staticServerHostnameID, staticServerIPID)
+
+			// Deploy the static client
+			logger.Log(t, "deploying static client")
+			k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-tproxy")
+
+			staticServerIP, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "get", "po", "-l", "app=static-server", `-o=jsonpath={.items[0].status.podIP}`)
+			require.NoError(t, err)
+			require.NotEmpty(t, staticServerIP)
+
+			// TODO: add http protocol test when it's merged.
+			staticServerHostnameURL := fmt.Sprintf("https://%s", staticServerServiceName)
+			staticServerIPURL := fmt.Sprintf("http://%s", staticServerIP)
+
+			// Create the service default declaring the external service (aka Destination)
+			createServiceDefaultDestination(t, consulClient, "", staticServerHostnameID, "", 443, staticServerServiceName)
+			createServiceDefaultDestination(t, consulClient, "", staticServerIPID, c.protocol, 80, staticServerIP)
```

Comment:
I'm not sure I understand the question @thisisnotashwin. Can you give an example of what other places we are supposed to configure?

Reply:
I suspect the suggestion is something like this to reduce the test cases.
```suggestion
			createServiceDefaultDestination(t, consulClient, "", "sd-http", "http", 80, staticServerIP)
			createServiceDefaultDestination(t, consulClient, "", "sd-tcp", "tcp", 80, staticServerIP)
```
But AFAIK we can't do this because the envoy `filterChainMatchers` will be the same.

Reply:
yup!! what @DanStough said! That makes sense. thanks so much for the clarification!

Reply:
If that is the case, can we create the servicedefault destination with the first protocol, make an assertion, and then update it with the other protocol and make the same assertion? I ask because deployments are expensive in terms of the acceptance test runtimes, and the more assertions we can squeeze into fewer installations, the better it is for the tests in the long run.